### DISCUSSION TITLE
A few newbish questions.

### DISCUSSION THREAD

**Discussion by jdubansky** [openstates/issues#191] [Q&A] [answered]
Hello!

I love what yall are doing here and the information is incredibly valuable to me for a project I am working on. I am trying to get everything setup on my end given the directions in the docs and on the READMEs but running into some issues. If I am understanding how all of these services connect properly, I need to have the docker-compose running from repo 'openstates.org' for the other services (like openstates-scrapers) to add to the database? 

I also cant seem to get the openstates-geo running at all. When I run the poetry command i get back "ModuleNotFoundError: No module named 'requests'" and when I try the docker command it says "make-tiles_1  | /usr/bin/env: ‘python’: No such file or directory"

**Comment by jamesturk** [comment, marked as answer]
Welcome!

Yes, the openstates.org project starts a shared database the others can access.  There are ways (as noted in the docs) to run without the database if you just want to work on scrapers/etc. 

re:openstates-geo, I'll take a look at it's docker setup, I believe it is likely just a bit out of date.  The project mainly exists to convert shapefiles into mbtiles though, so unless you need to do that for your own mapbox maps you can usually safely ignore it.

**Reply by jamesturk** [reply]
PS just saw your other email, if you have bandwidth and would like to try to get the geo repo working that'd be great, but it isn't one that most people work with/in, so depending on your needs/etc. it is fine to leave as-is and we'll get to a fix next time it is necessary (which is quite rarely, generally just when redistricting happens)

**Reply by jdubansky** [reply]
Thanks for the quick reply! I had some time to play around with it and im a bit closer to getting it to work I think. Is there a copy of the mbtiles that is available in bulk somewhere?

**Comment by jamesturk** [comment]
Not right now, I could probably generate one sometime soon, what're you
using it for if I can ask?

**Reply by jdubansky** [reply]
I was able to get it up and running! was missing the `poetry install` step before launching the other scripts. I also found a few errors in the scripts but dont have permissions to write to the repo. I wanted the ability to have similar maps that are on the openstates website.

**Comment by jamesturk** [comment]
Would you be able to submit a PR?

i’d be glad to look at any fixes

**Reply by jdubansky** [reply]
it doesnt look like I have permissions to push a branch to the repo. Is that something you need to enable on your end?

**Comment by jamesturk** [comment]
The typical flow is to fork the repo, and issue a PR.  There’s a good guide on it here:


https://docs.github.com/en/free-pro-team@latest/github/getting-started-with-github/fork-a-repo

**Reply by jdubansky** [reply]
Gotcha! let me do some reading on this and try it out today! sorry for the newbish questions 🤦

**Reply by jdubansky** [reply]
Sooo I think I did it correctly..

https://github.com/openstates/openstates-geo/compare/main...jdubansky:setup?expand=1
